File: 1/2021.04.14/452.Minimum Number of Arrows to Burst Balloons/java/solution.java
class Solution {
    public int findMinArrowShots(int[][] points) {
        Arrays.sort(points, (a, b) -> {
            return a[1] > b[1] ? 1 : -1;
        });
        int count = 1;
        int end = points[0][1];
        for (int[] point : points) {
            int start = point[0];
            if (start > end) {
                count++;
                end = point[1];
            }
        }
        return count;
    }
}

File: 1/2021.04.14/452.Minimum Number of Arrows to Burst Balloons/python/solution.py
class Solution:
    def findMinArrowShots(self, points: List[List[int]]) -> int:
        points.sort(key=lambda x : x[1])
        count = 1
        end = points[0][1]
        for interval in points:
            start = interval[0]
            if start > end:
                count += 1
                end = interval[1]
        return count
